cc3d/player5/Plugins/ViewManagerPlugins/PlotWindowInterface.py

```diff
@@ -169,13 +154,44 @@ def set_data_bar_graph_item(self, plot_obj, x, y):
     def addPlot(self, _plotName, _style="Lines", _color='white', _size=3, _alpha=255):
         return self.add_plot(plot_name=_plotName, style=_style, color=_color, size=_size, alpha=_alpha)
 
-    def add_plot(self, plot_name, style="Lines", color='white', size=3, alpha=255):
+    def add_plot(self, plot_name, style="Lines", color='white', size=3, alpha=255,
+                 separate_y_axis=False, y_min=None, y_max=None, y_scale_type=None):
+        """
+        User's API function - adds a data series plot to the plotting window
+        :param plot_name:
+        :param style:
+        :param color:
+        :param size:
+        :param alpha:
+        :param separate_y_axis:
+        :param y_min:
+        :param y_max:
+        :param y_scale_type:
+        :return:
+        """
+
+        if not separate_y_axis:
+            if y_min is not None or y_max is not None:
+                raise RuntimeError('y_min or y_max can only be set if you set separate_y_axis=True '
```

Comment:
Is there any technical reason for this? If not I vote for letting the user set limits

Reply:
you cannot be setting different limits for different data sets if you are using only single y_axes. It would make no sense to have first data series have limits 0..1 and second data series (on the same plot with shared y_axis) have limit from 2..3. Because this should would you choose then?

Reply:
I may have misunderstood the function then. 

Right now the user can set fixed axis range in player by right clicking the plot and setting the y/x range, my assumption was that this (without the raise error) could be used to do the same from python (I can't recall if that's already possible)

Reply:
True (about right-click) but this PR lets you do a bit more programmatically and having multiple axes was not available before.

Reply:
>  lets you do a bit more programmatically

That was what I was getting at, (if I understood the code correctly) this PR wouldn't allow the user to set the y limits from python if they are doing a single axis but would allow it for multiple.

Reply:
Exactly, I think this is good. Let me know if you agree or if we should change the behavior (we can always change the behavior later as well)